pkg/dbsyncers/policy_syncer.go

```diff
@@ -149,10 +151,87 @@ func updateComplianceStatus(ctx context.Context, k8sClient client.StatusClient,
 		policy.Status.ComplianceState = policiesv1.Compliant
 	}
 
-	err := k8sClient.Status().Patch(ctx, policy, client.MergeFrom(originalPolicy))
+	placements, err := getPlacements(k8sClient, policy)
+	if err != nil {
+		return err
+	}
+	policy.Status.Placement = placements
+
+	err = k8sClient.Status().Patch(ctx, policy, client.MergeFrom(originalPolicy))
 	if err != nil && !errors.IsNotFound(err) {
 		return fmt.Errorf("failed to update policy CR: %w", err)
 	}
 
 	return nil
 }
+
+func getPlacements(k8sClient client.Client, instance *policiesv1.Policy) ([]*policiesv1.Placement, error) {
+	log.Info("get the placements for the policy", "policy", instance.GetName())
+	pbList, err := getPlacementBindingList(k8sClient, instance)
```

Comment:
this implementation is very wasteful.
think about the following example - 
let's assume we have 100 policies in the same namespace. this code will get all the bindings in the namespace 100 times.
that means that if we have 100 bindings, we get each 100 times - the code will get bindings 10000 times with no good reason (!).

I suggest to change this. we may use a cache in memory for bindings per namespace.
when we handle a policy we can check in the cache if we already read the bindings in the ns and then we don't need to get it again. in every sync cycle we can build the cache from scratch for simplicity.  

this will reduce the number of bindings we get in the above example to 100.

Reply:
I agree with your point that this implementation is very wasteful. That is existing logic in `governance-policy-propagator`. I am thinking it carefully. I still think if the policy is a global policy, then it should be handled by HoH controller. for the long term, the hoh controller can watch the placementbinding to update the policy.  we need to introduce the mechanism to update the cache if we cache the placementbindings.

for the short term, just use this solution to update the placements section. WDYT?

Reply:
I'm not sure this will work good.
we want status sync to be responsive and therefore we configured it to have a sync interval of 5 seconds (5 seconds between consecutive syncs).

if we're talking about few bindings, it will probably work fine.
if we take the above example - there is no chance that status sync will manage to read 10K resources in 5 seconds cycle and we might get to a non responsive status sync (if sync didn't finish in the interval context might get cancelled and nothing is updated. 
see here - https://github.com/stolostron/hub-of-hubs-status-sync/blob/main/pkg/dbsyncers/generic_db_syncer.go#L52-L54)

Reply:
I noticed this problem existed in the current implementation. any concerns to add the logic to update placement status in policy CR in spec-sync? in spec-sync, we watch the placementbinding, so I think that is a proper place to do it. And we can keep the policy spec to have placement status only in CR and DB. I mean do not clean up the `placement` in status. we can keep it to sync-up to the leaf hub. It should not any impacts.

Reply:
I do not want to put it in governance-policy-propagator due to:
1. if the policy is global policy, it should be handled by HoH controller.
2. It can be easy to adopt by policy squad via filtering out the global policy in mapper. e.g.: https://github.com/clyang82/governance-policy-propagator/blob/5844672aff73fe4915ae54020a3ca5d32600296a/controllers/propagator/policyMapper.go#L58
Does it make sense?

Reply:
yes. it makes sense.
let's take the option to put it in policy propagator off the table.
so we're left only with the option to handle global policies in one of the HoH controllers.

regarding the above suggestion - this violates the separation of concerns we kept in the design and is mixing spec/status updates. I don't think this is a good idea.
the clean separation allows us flexibility and customization that this change may break.
let me think about alternative approach.

Reply:
@nirrozenbaum please take a look at this approach - https://github.com/stolostron/hub-of-hubs-manager/pull/15

Reply:
@clyang82 this approach makes a lot of sense.
+1 from me on this.

wrote two minor comments in hub-of-hubs-manager PR.

Reply:
Thanks @nirrozenbaum let me close this PR. trace it in https://github.com/stolostron/hub-of-hubs-manager/pull/15